spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/FilterNotFoundException.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.feature.manager;
+
+import com.microsoft.azure.spring.cloud.feature.manager.entities.FeatureFilterEvaluationContext;
+
+/**
+ * This class defines a custom exception type for when an expected Filter is not found
+ * when checking if a Feature is enabled. A FilterNotFoundException is only thrown when
+ * failfast is enabled, which is true by default.
+ *
+ */
+public class FilterNotFoundException extends Exception {
+
+    private static final long serialVersionUID = 1L;
+
+    private final FeatureFilterEvaluationContext filter;
+
+    /**
+     * Creates a new instance of the FilterNotFoundException
+     * 
+     * @param message the error message.
+     * @param cause the original error thrown, typically of NoSuchBeanDefinitionException type.
+     * @param filter The filter context used to find the not found filter.
+     */
+    public FilterNotFoundException(String message, Throwable cause, FeatureFilterEvaluationContext filter) {
+        super(message, cause);
+        this.filter = filter;
+    }
+
+    @Override
+    public String getMessage() {
+        if (filter == null) {
+            return getCause().getMessage();
+        }
+        return getCause().getMessage() + ", " + filter.toString();
```

Comment:
nit: is it showing the `filter.toString()` only in the message a little hard for the user to understand the error message?